jax/_src/numpy/lax_numpy.py

```diff
@@ -1370,21 +1371,22 @@ def nonzero(a: ArrayLike, *, size: int | None = None,
   arr = atleast_1d(a)
   del a
   mask = arr if arr.dtype == bool else (arr != 0)
-  if size is None:
-    size = mask.sum()
-  size = core.concrete_dim_or_error(size,
+  calculated_size = mask.sum() if size is None else size
+  calculated_size = core.concrete_dim_or_error(calculated_size,
     "The size argument of jnp.nonzero must be statically specified "
     "to use jnp.nonzero within JAX transformations.")
-  if arr.size == 0 or size == 0:
-    return tuple(zeros(size, int) for dim in arr.shape)
-  flat_indices = reductions.cumsum(bincount(reductions.cumsum(mask), length=size))
-  strides = (np.cumprod(arr.shape[::-1])[::-1] // arr.shape).astype(int_)
+  assert isinstance(calculated_size, int)
```

Comment:
Actually, sorry this can also be a dynamic dimension size, so this will probably break some tests.

I don't want to annotate `jnp.bincount` to accept Arrays though, so probably the best thing to do is to use `type: ignore`

Reply:
Right, that's what I figured. I'll do that.

Reply:
I imagine you'll eventually add an annotation for dynamic dimension sizes?  You could add something like this for now?
```
DimSize: TypeAlias = int
```

Reply:
No, `DimSize` is only partially implemented all across the package. The TODO is implicit everywhere 😁

Reply:
Got it!